src/hotspot/share/gc/serial/serialFullGC.cpp

```diff
@@ -96,17 +96,16 @@ class DeadSpacer : StackObj {
   ContiguousSpace* _space;
 
 public:
-  DeadSpacer(ContiguousSpace* space) : _allowed_deadspace_words(0), _space(space) {
-    size_t ratio = _space->allowed_dead_ratio();
-    _active = ratio > 0;
+  DeadSpacer(ContiguousSpace* space, size_t allowed_dead_ratio) : _allowed_deadspace_words(0), _space(space) {
+    _active = allowed_dead_ratio > 0;
```

Comment:
How about

```
    size_t ratio = (_space == SerialHeap::heap()->old_gen()->space())
                 ? MarkSweepDeadRatio
                 : 0;
```
?

Reply:
If each space have a `allowed_dead_ratio`, it is easy to modify one of them and meet the changeable demand in the future. But these years, we rarely have a demand to revise them. Maybe the previous code and this current patch are both over-designed. Fixed just as you suggested.